### PR TITLE
CircleCI setup

### PR DESCRIPTION
**Pull request by LukasPaczos**
This PR adds basic CircleCI setup.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+checkstyle:
+	./gradlew checkstyle
+
+test:
+	./gradlew :library:test
+	./gradlew :library:test
+
```
Comment (tobrun) [inline Makefile]: Double invocation?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+release:
+	./gradlew :library:assembleRelease
+	./gradlew :library:assembleRelease
```
Comment (tobrun) [inline Makefile]: Double invocation?

File: circle.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+workflows:
+  version: 2
+  default:
+    jobs:
+      - build
+      - release
+jobs:
+  build:
+    working_directory: ~/code
```
Comment (tobrun) [inline circle.yml]: Is this line required?

File: library/build.gradle [hunk 4]
```diff
@@ -1,18 +1,26 @@
 apply plugin: 'com.android.library'
 
-android {
-    compileSdkVersion 26
-
+version = VERSION_NAME
```
Comment (tobrun) [inline library/build.gradle]: Are we using version on mutliple places? Only seeing it used for versionName

File: circle.yml (hunk 3 above)
Comment (LukasPaczos) [reply circle.yml]: It was captured from our implementation in [events repo](https://github.com/mapbox/mapbox-events-android/blob/74c1e851277530cb4531db861486fc030cd484a6/.circleci/config.yml#L11). I don't know whether there are any benefits to this approach.

File: app/build.gradle [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        }
+    }
+
+    testOptions {
+        unitTests.returnDefaultValues = true
+    }
+}
+
+dependencies {
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
```
Comment (zugaldia) [inline app/build.gradle]: probably safe to remove.

File: build.gradle [hunk 6]
```diff
@@ -22,6 +21,11 @@ allprojects {
     }
 }
 
+subprojects {
+    apply plugin: 'idea'
+    apply from: "${rootDir}/gradle/dependencies.gradle"
```
Comment (zugaldia) [inline build.gradle]: Do we need to invoke `apply from: "${rootDir}/gradle/dependencies.gradle"` twice in this file?

File: library/build.gradle [hunk 7]
```diff
@@ -22,13 +28,19 @@ android {
         }
     }
 
+    testOptions {
+        unitTests.returnDefaultValues = true
+    }
 }
 
 dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
```
Comment (zugaldia) [inline library/build.gradle]: We can also remove this line.

File: gradle.properties [hunk 8]
```diff
@@ -1,17 +1,30 @@
...
-# Gradle settings configured through the IDE *will override*
-# any settings specified in this file.
+POM_URL=https://github.com/mapbox/mapbox-gestures-android
+POM_SCM_URL=https://github.com/mapbox/mapbox-gestures-android
+POM_SCM_CONNECTION=scm:git@github.com:mapbox/mapbox-gestures-android.git
+POM_SCM_DEV_CONNECTION=scm:git@github.com:mapbox/mapbox-gestures-android.git
 
-# For more details on how to configure your build environment visit
-# http://www.gradle.org/docs/current/userguide/build_environment.html
+POM_LICENCE_NAME=The Apache Software License, Version 2.0
```
Comment (zugaldia) [inline gradle.properties]: License is actually BSD - https://github.com/mapbox/mapbox-gestures-android/blob/master/LICENSE.md.

**Comment by LukasPaczos** [comment]
Ready for another round @zugaldia, @tobrun.

**Pull request merged**